runtime/util_core/j9argscan.c

```diff
@@ -385,6 +385,102 @@ scan_hex_caseflag_u64(char **scan_start, BOOLEAN uppercaseAllowed, uint64_t* res
 	return bits;
 }
 
+/**
+ * Scan the next unsigned number off of the argument string, and parses for memory sizes.
+ * Specify the size in TiBs, GiBs, MiBs, or KiBs (with T,t,G,g,M,m,K,k suffixes) or in bytes (no suffix).
+ * @param[in] scan_start The string to be scanned
+ * @param[out] result The result
+ * @return returns 0 on success, 1 if the argument string is not a number, or 2 if overflow occurs.
+ */
+uintptr_t
+scan_u64_memory_size(char **scan_start, uint64_t* result)
+{
+	uintptr_t rc = scan_u64(scan_start, result);
+
+	/* Report any errors from scan */
+	if (0 != rc) {
+		return rc;
+	}
+
+	if (try_scan(scan_start, "T") || try_scan(scan_start, "t")) {
+		if (*result <= (((U_64)-1) >> 40)) {
+			*result <<= 40;
+		} else {
+			rc = 2;
+		}
+	} else if (try_scan(scan_start, "G") || try_scan(scan_start, "g")) {
+		if (*result <= (((U_64)-1) >> 30)) {
+			*result <<= 30;
+		} else {
+			rc = 2;
+		}
+	} else if (try_scan(scan_start, "M") || try_scan(scan_start, "m")) {
+		if (*result <= (((U_64)-1) >> 20)) {
+			*result <<= 20;
+		} else {
+			rc = 2;
+		}
+	} else if (try_scan(scan_start, "K") || try_scan(scan_start, "k")) {
+		if (*result <= (((U_64)-1) >> 10)) {
+			*result <<= 10;
+		} else {
+			rc = 2;
+		}
+	}
+	return rc;
+}
+
+/**
+ * Scan the next unsigned number off of the argument string, and parses for memory sizes.
+ * Specify the size in TiBs, GiBs, MiBs, or KiBs (with T,t,G,g,M,m,K,k suffixes) or in bytes (no suffix).
+ * @param[in] scan_start The string to be scanned
+ * @param[out] result The result
+ * @return returns 0 on success, 1 if the argument string is not a number, or 2 if overflow occurs.
+ */
+uintptr_t
+scan_udata_memory_size(char **scan_start, uintptr_t* result)
+{
+	uintptr_t rc = scan_udata(scan_start, result);
+
+	/* Report any Errors from Scan */
+	if (0 != rc) {
+		return rc;
```

Comment:
you can eliminate this return too... just use `if (0 == rc) {` to wrap `if` statement below